src/Symfony/Component/Validator/Constraints/Composite.php

```diff
@@ -83,6 +83,13 @@ public function __construct($options = null)
             $mergedGroups = [];
 
             foreach ($nestedConstraints as $constraint) {
+                if ($constraint instanceof self && !$constraint->groups) {
```

Comment:
Why doing this here and not below?

I mean something like this:

```php
$this->groups = array_keys($mergedGroups);

if (!$this->groups) {
    $this->groups = [self::DEFAULT_GROUP];
}
```

Reply:
I guess this is equivalent, I've updated for a ternary because it's simpler indeed.